packages/strapi-plugin-users-permissions/controllers/User.js

```diff
@@ -122,4 +123,22 @@ module.exports = {
 
     ctx.body = sanitizeUser(user);
   },
+
+  /**
+   * Update authenticated user.
+   * @return {Object|Array}
+   */
+  async updateMe(ctx) {
+    const input = ctx.request.body;
+
+    try {
+      await validateProfileUpdateInput(input);
+    } catch (err) {
+      return ctx.badRequest('ValidationError', err);
+    }
+
+    const updatedUser = await strapi.plugins['users-permissions'].services.user.edit(ctx.state.user.id, input);
```

Comment:
Incase some one needs this 
Change - `services.user.edit({ id: ctx.state.user.id }, input);`

Reply:
I wouldn't recommend using this yet, it's only a prototype with no validation or security yet.

Reply:
Sure
This Pointed me in the right direction , i have added my own validations